### PR TITLE
Refactor/log format

### PR DESCRIPTION
**Pull request by SamuelRioTz**
change log from `.txt` to `.json`

### CODE REVIEW

File: src/OSM_dataTool.js [hunk 1]
```diff
@@ -20,7 +19,7 @@ module.exports = function ({ routes, ways, assumeFirstWayIsStart, mapProperties,
             const data = routeExtractor(current_route, ways, assumeFirstWayIsStart)
 
             routes_complete++
-            log_file += `\nDone >>> ${name}`
+            log_file.push(Object.assign({ id: current_route.id }, current_route.tags))
```
Comment (mdoelker) [inline src/OSM_dataTool.js]: `id` might be overridden, if tags has an `id` key.

```suggestion
            log_file.push({ id: current_route.id, tags: current_route.tags })
```

File: src/OSM_dataTool.js [hunk 2]
```diff
@@ -45,13 +44,20 @@ module.exports = function ({ routes, ways, assumeFirstWayIsStart, mapProperties,
                 }
             })
         } catch (error) {
-            debug(`Error: ${error.message}`)
+            debug(`Error: ${error.extractor_error || error.message}`)
+            log_file.push(Object.assign({
+                id: current_route.id,
+                error_log: error.extractor_error ? error : "not controlled"
```
Comment (mdoelker) [inline src/OSM_dataTool.js]: Isn't `error` a more concise name?

```suggestion
                error: error.extractor_error ? error : "not controlled"
```

File: src/extractor_error.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+    duplicated: "duplicated",
+    nomergeable: "nomergeable",
```
Comment (mdoelker) [inline src/extractor_error.js]: No space here?

```suggestion
    not_mergeable: "not mergeable",
```

**Comment by mdoelker** [comment]
Is this ready for merging?

**Comment by SamuelRioTz** [comment]
@mdoelker now is ready

**Pull request merged**